### PR TITLE
User mute expiration

### PR DESCRIPTION
**Pull request by gumuz**
Added options to `MuteUser` and `MuteUsers` to support the recently added `timeout` option for Mute expiration

### CODE REVIEW

**Comment by gumuz** [comment]
Not sure if it fails because of a bug or some configuration problem. @ferhatelmas is this test dependent on the fixes you were preparing for the chat api?

File: user.go [hunk 1]
```diff
@@ -77,39 +78,37 @@ func (u User) MarshalUnknowns(out *jwriter.Writer, first bool) {
...
 		return errors.New("target IDs are empty")
 	case userID == "":
 		return errors.New("user ID is empty")
 	}
 
-	data := map[string]interface{}{
-		"target_ids": targetIDs,
-		"user_id":    userID,
-	}
+	options["target_ids"] = targetIDs
```
Comment (ferhatelmas) [inline user.go:108]: `options` can be nil here

File: user_test.go [hunk 2]
```diff
@@ -30,31 +30,41 @@ func TestClient_MuteUser(t *testing.T) {
...
 	assert.NotEmpty(t, mute.Target, "mute has target")
+	assert.NotEmpty(t, mute.Expires, "mute has expires")
 }
 
 func TestClient_MuteUsers(t *testing.T) {
 	c := initClient(t)
 	initChannel(t, c)
 
-	users := []string{randomUser().ID, randomUser().ID}
+	usernames := []string{randomUser().ID, randomUser().ID}
```
Comment (ferhatelmas) [inline user_test.go:54]: this randomness can cause issues in tests once in a while, plug in some specific users

File: user_test.go [hunk 3]
```diff
@@ -30,31 +30,41 @@ func TestClient_MuteUser(t *testing.T) {
...
+	err := c.MuteUsers(usernames, serverUser.ID, map[string]interface{}{"timeout": 60})
 	require.NoError(t, err, "mute user")
+
+	users, err := c.QueryUsers(&QueryOption{
+		Filter: map[string]interface{}{
+			"id": map[string]string{"$eq": serverUser.ID},
+		}})
+	require.NoError(t, err, "query users")
+
+	for _, mute := range users[0].Mutes {
```
Comment (ferhatelmas) [inline user_test.go:65]: ensure length before, it can panic here otherwise

File: query.go [hunk 4]
```diff
@@ -81,8 +80,6 @@ func (c *Client) QueryUsers(q *QueryOption, sorters ...*SortOption) ([]*User, er
 		return nil, err
 	}
 
-	fmt.Println(string(data))
```
Comment (ferhatelmas) [inline query.go:84]: :+1:

**Comment by ferhatelmas** [comment]
Put some comments, see them. 

However, it is failing because tests put some garbage and didn't clean it so it's at size limit of some resources. I will clean them and additionally we should update tests to clean up themselves.

**Comment by ferhatelmas** [comment]
I did some cleanup in its app, should be fine now.

File: user_test.go [hunk 5]
```diff
@@ -28,33 +28,61 @@ func TestClient_MuteUser(t *testing.T) {
 	c := initClient(t)
 	initChannel(t, c)
 
-	user := randomUser()
+	assert.GreaterOrEqualf(t, len(testUsers), 1, "not enough test users: %+v", testUsers)
```
Comment (ferhatelmas) [inline user_test.go:31]: it won't stop the test if fails so you can still have panics so if it's a hard requirement like length check, prefer `require`.

File: user_test.go [hunk 6]
```diff
@@ -28,33 +28,63 @@ func TestClient_MuteUser(t *testing.T) {
...
+	assert.NotEmpty(t, mute.Target, "mute should have a Target")
+	assert.NotEmpty(t, mute.Expires, "mute should have Expires")
 }
 
 func TestClient_MuteUsers(t *testing.T) {
 	c := initClient(t)
 	initChannel(t, c)
 
-	users := []string{randomUser().ID, randomUser().ID}
+	assert.GreaterOrEqualf(t, len(testUsers), 2, "not enough test users: %+v", testUsers)
```
Comment (ferhatelmas) [inline user_test.go:72]: same logic should be here too

Comment (gumuz) [reply user_test.go:72]: I was just about to push that now!

Comment (ferhatelmas) [reply user_test.go]: ah, sorry, I thought it was ready

**Comment by ferhatelmas** [comment]
@gumuz Do you want to release it ?

* changelog
* bump version 
* tag (minor in this case)

**Pull request merged**